CMakeLists.txt

```diff
@@ -57,3 +67,105 @@ if(WITH_TESTS)
   enable_testing()
   add_subdirectory(test)
 endif()
+
+set(CPACK_PACKAGE_NAME cpp-dependencies)
+set(CPACK_PACKAGE_VENDOR "TomTom International BV")
+set(CPACK_PACKAGE_CONTACT "${CPACK_PACKAGE_VENDOR}")
+set(CPACK_DEBIAN_PACKAGE_SHLIBDEPS ON)
+
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "Tool to check C++ #include dependencies
+ The tool cpp-dependencies creates #include dependency information for C++
+ source code files, which it derives from scanning a full source tree.
+ .
+ The dependency information is output as .dot files, which can be visualized
+ in, for example, GraphViz.")
+set(CPACK_DEBIAN_PACKAGE_HOMEPAGE https://github.com/tomtom-international/cpp-dependencies)
+set(CPACK_DEBIAN_PACKAGE_SUGGESTS graphviz)
+set(CPACK_DEBIAN_PACKAGE_SECTION devel)
+
+# Version: update this before or while tagging a new release, for out of repo builds
+set(CPACK_PACKAGE_VERSION_MAJOR 1)
+set(CPACK_PACKAGE_VERSION_MINOR 1)
+set(CPACK_PACKAGE_VERSION_PATCH 0)
+
+# Automatically detect package version to use from git
+find_package(Git)
+if(Git_FOUND OR GIT_FOUND)
+  execute_process(
+    COMMAND ${GIT_EXECUTABLE} describe --tags --long --dirty --always
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    RESULT_VARIABLE DESCRIBE_RESULT
+    OUTPUT_VARIABLE DESCRIBE_STDOUT
+    )
+  if(DESCRIBE_RESULT EQUAL 0)
+    string(STRIP "${DESCRIBE_STDOUT}" DESCRIBE_STDOUT)
+    message(STATUS "Git reported this project's version as '${DESCRIBE_STDOUT}'")
+    if(DESCRIBE_STDOUT MATCHES "^(.*)-(dirty)$")
+      set(DESCRIBE_DIRTY "${CMAKE_MATCH_2}")
+      set(DESCRIBE_STDOUT "${CMAKE_MATCH_1}")
+    endif()
+    if(DESCRIBE_STDOUT MATCHES "^([0-9a-f]+)$")
+      set(DESCRIBE_COMMIT_NAME "${CMAKE_MATCH_1}")
+      set(DESCRIBE_STDOUT "")
+    elseif(DESCRIBE_STDOUT MATCHES "^(.*)-g([0-9a-f]+)$")
+      set(DESCRIBE_COMMIT_NAME "${CMAKE_MATCH_2}")
+      set(DESCRIBE_STDOUT "${CMAKE_MATCH_1}")
+    endif()
+    if(DESCRIBE_STDOUT MATCHES "^(.*)-([0-9]+)$")
+      set(DESCRIBE_COMMIT_COUNT "${CMAKE_MATCH_2}")
+      set(DESCRIBE_TAG "${CMAKE_MATCH_1}")
+      set(DESCRIBE_STDOUT "")
+    endif()
+    if("${DESCRIBE_TAG}.0.0" MATCHES "^([0-9]+)\\.([0-9]+)\\.([0-9]+).*$")
+      set(CPACK_PACKAGE_VERSION_MAJOR "${CMAKE_MATCH_1}")
+      set(CPACK_PACKAGE_VERSION_MINOR "${CMAKE_MATCH_2}")
```

Comment:
No, it adds the ".0.0" to the end that a tag of "1" becomes parsed as "1.0.0", but "1.1" becomes "1.1.0.0" where the regex just ignores the last ".0". I.e. as input this supports version numbers of form "x", "x.y", "x.y.z" and will match "x.y.z.u" but it will ignore the "u" part (so while matched those are not supported).